public/pages/Dashboard/Container/DashboardOverview.tsx

```diff
@@ -237,28 +258,31 @@ export function DashboardOverview(props: OverviewProps) {
     filterSelectedDetectors(
       selectedDetectorsName,
       selectedDetectorStates,
-      selectedIndices,
+      selectedIndices
     );
   }, [selectedDetectorsName, selectedIndices, selectedDetectorStates]);
 
-  const DataSourceMenu =
-    getDataSourceManagementPlugin().ui.getDataSourceMenu<DataSourceSelectableConfig>();
-  const renderDataSourceComponent = useMemo(() => {
-    return (
-      <DataSourceMenu
-        setMenuMountPoint={props.setActionMenu}
-        componentType={'DataSourceSelectable'}
-        componentConfig={{
-          fullWidth: false,
-          activeOption:[{ id: MDSOverviewState.selectedDataSourceId }],
-          savedObjects: getSavedObjectsClient(),
-          notifications: getNotifications(),
-          onSelectedDataSources: (dataSources) =>
-            handleDataSourceChange(dataSources),
-        }}
-      />
-    );
-  }, [getSavedObjectsClient(), getNotifications(), props.setActionMenu]);
+  let renderDataSourceComponent = null;
+  if (dataSourceEnabled) {
+    const DataSourceMenu =
+      getDataSourceManagementPlugin().ui.getDataSourceMenu<DataSourceSelectableConfig>();
```

Comment:
Can this trigger an NPE?

Reply:
I understand the datasourcenabled check has passed here, but that is from a different plugin. prefer to have this more safely handled

Reply:
I understand that's the current mechanism. But given the fact this is technically an optional plugin, let's add a single `?` check to prevent NPE.

Reply:
updated in the latest revision :)